pkg/integrations/mongodb_exporter/mongodb_exporter.go

```diff
@@ -17,15 +17,29 @@ import (
 
 var DefaultConfig = Config{
 	DirectConnect: true,
+	EnableDBStats: false,
```

Comment:
wouldn't be better to have all of this enabled by default instead of setting them to true later (from line 92).

Reply:
Maybe I'm missing something, but I still don't understand why these flags are not enabled by default and we need a `collectAll` variable (which is the default behavior).

Reply:
Yeah it is bit confussing but setting collectAll to true and enabling all other other flags doesn't make any differnce. In the comment i added for the sake of understanding, collectAll=ture is equaliant to enabling all the other flags  however, if we enable any one or more flags it will diasable collectAll and other flags too.

Reply:
If it's confusing for you, it will be confusing for the rest of users. Would you mind refactor the configuration to be easier to understand?

Reply:
I think I understand the point that, because `CollectAll` was hardcoded to `true`, the new flags won't have any effect unless we expose the flag and users manually disable it.
 
In order to mirror the actual [collector's logic](https://github.com/percona/mongodb_exporter/blob/cb3db8922fb068bba6ecd4c12c231cb63d970515/exporter/exporter.go#L153-L165) as closely as possible and avoid misunderstanding from users, I would suggest fully exposing `CollectAll` as a new flag, and setting it to `true` in `DefaultConfig` (as @marctc suggested).

Rationale: most users won't need that level of granularity, but if they do, they can disable `CollectAll` and set each desired flag individually.

We should also make sure that we are exposing all the flags that would be enabled by `CollectAll` (afaik we are missing [some](https://github.com/percona/mongodb_exporter/blob/cb3db8922fb068bba6ecd4c12c231cb63d970515/exporter/exporter.go#L157-L165)).

This is also a potential argument for not exposing any flags at all: if the collector dependency is updated and some new flags are added and missed as part of the update, users who manually set the list of collectors to enable may miss the new ones, so there is some maintenance cost involved in providing such level of granularity.